test/contracts/OVM/verification/OVM_FraudVerifier.spec.ts

```diff
@@ -178,6 +178,54 @@ describe('OVM_FraudVerifier', () => {
           ).to.equal(Mock__OVM_StateTransitioner.address)
         })
       })
+
+      describe('when provided an invalid transaction proof index', () => {
+        before(() => {
+          Mock__OVM_CanonicalTransactionChain.smocked.verifyTransaction.will.return.with(
+            false
+          )
+        })
+
+        it('should revert', async () => {
+          let transactionDummyBatchProof = DUMMY_BATCH_PROOFS[0];
+          let preStateDummyBatchProof = DUMMY_BATCH_PROOFS[0];
+          transactionDummyBatchProof.index = 11;
+          preStateDummyBatchProof.index = 3;
+          await expect(
+            OVM_FraudVerifier.initializeFraudVerification(
+              NULL_BYTES32,
+              DUMMY_BATCH_HEADERS[0],
+              preStateDummyBatchProof,
+              DUMMY_OVM_TRANSACTIONS[0],
+              DUMMY_TX_CHAIN_ELEMENTS[0],
+              DUMMY_BATCH_HEADERS[0],
+              transactionDummyBatchProof
+            )
+          ).to.be.revertedWith('Invalid transaction inclusion proof.')
+        })
+      })
+    })
```

Comment:
I think that instead of this test, what we're looking for are tests that get `revertedWith("Pre-state root global index must equal to the transaction root global index")`, since we've added that assertion in the contract. So this would be the case where the inclusion proofs for the state root and the transaction both pass, but they are for non-matching indices.